### PR TITLE
[BSE-40] Add case sensitivity option to the like quantify operators

### PR DESCRIPTION
**Pull request by jsternberg**
This adds case sensitivity to the quantify operator for `LIKE` and adds
new operators to the standard operator table and teaches the
`SqlToRelConverter` how to expand case insensitive like queries to use
`ILIKE`.

### CODE REVIEW

**Comment by jsternberg** [comment]
@keatoooon for the interim we're just going to hack some support for ILIKE into the fork. In the future, we can discuss refactoring this aspect to move the function calls outside of the fork and handle the rewrite with planner rules instead and see if that works. But, I think it's really not worth doing this at this time since it would mean that I have to move everything over or move only half of it over which would be even more confusing.

File: core/src/main/java/org/apache/calcite/sql2rel/SqlToRelConverter.java [hunk 1]
```diff
@@ -1778,8 +1789,14 @@ public RelNode convertToSingleValueSubq(
...
-      comparisonOp = RelOptUtil.op(((SqlQuantifyOperator) op).comparisonKind,
-          SqlStdOperatorTable.EQUALS);
+      SqlQuantifyOperator quantifyOperator = (SqlQuantifyOperator) op;
+      if (quantifyOperator.comparisonKind == SqlKind.LIKE && !quantifyOperator.caseSensitive) {
+        // Special code for case insensitive like.
+        comparisonOp = SqlLibraryOperators.ILIKE;
+      } else {
+        comparisonOp = RelOptUtil.op(quantifyOperator.comparisonKind,
+            SqlStdOperatorTable.EQUALS);
+      }
```
Comment (keatoooon) [inline core/src/main/java/org/apache/calcite/sql2rel/SqlToRelConverter.java:1799]: I think this is the only actual change to this file, the rest is formatting. If there is another change I need to review, please let me know.

**Review by keatoooon — APPROVED** [review]
I'm ok with the hack, LGTM

File: core/src/main/java/org/apache/calcite/sql2rel/SqlToRelConverter.java (hunk 1 above)
Comment (jsternberg) [reply core/src/main/java/org/apache/calcite/sql2rel/SqlToRelConverter.java:1799]: This was the only one. I'm not really sure why IntelliJ feels the need to reformat this entire file when I touch it. I set the options to the ones for Calcite but I guess it'll just do what it does.

**Pull request merged**